### DISCUSSION TITLE
What is the recommended implementation approach when having multiple CRD versions?

### DISCUSSION THREAD

**Discussion by juangon** [operator-framework/java-operator-sdk#1710] [Q&A]
Hi again!

I am thinking of the approach to follow when having multiple CRD versions. What is the recommended approach? As I think 95% of the code (in reconcilers, etc) can be reused...could I just inherit the Reconcilers so they use the new versions and just change some few lines on some methods? Would that work? 

Thanks!

**Comment by csviri** [comment]
Yes, it can be reused (or should be, not sure if tested), just make sure that the actual reconciles have the `@ControllerConfiguration` annotation. and the proper custom resource type generics.

**Comment by metacosm** [comment]
Let us know if you have any issues, but yes, it should work, provided your reconcilers are properly typed and annotated.

**Comment by tombentley** [comment]
This would be a great addition to the website, or a sample.

**Reply by csviri** [reply]
there is already a sample: https://github.com/operator-framework/java-operator-sdk/blob/41463e128152f93d7c78745085d6eea1595ed2aa/operator-framework/src/test/java/io/javaoperatorsdk/operator/MultiVersionCRDIT.java.

**Reply by csviri** [reply]
In form of Integration test.

**Reply by csviri** [reply]
might be easier to brows the upcoming v5 structure:
https://github.com/operator-framework/java-operator-sdk/tree/e8d44c4df3edc32ffeb4a5c4f555a079862ede71/operator-framework/src/test/java/io/javaoperatorsdk/operator/baseapi/multiversioncrd
